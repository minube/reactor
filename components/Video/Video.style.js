import { StyleSheet } from 'react-native';

import { STYLE, THEME } from '../../common';

const { COLOR } = THEME;

export default StyleSheet.create({
  container: StyleSheet.flatten([
    STYLE.CENTERED,
    {
      alignSelf: 'center',
      overflow: 'hidden',
    },
  ]),

  loading: {
    backgroundColor: COLOR.BACKGROUND,
  },

  activity: {
    position: 'absolute',
    alignSelf: 'center',
  },

  webView: {
    flex: 1,
    overflow: 'hidden',
  },
});
