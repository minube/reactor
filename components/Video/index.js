import { ENV } from '../../common';
import Default from './Video';

const Loading = ENV.IS_NATIVE ? require('./Video.native').default : Default;

export default Loading;
