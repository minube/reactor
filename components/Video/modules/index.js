import embedUrl from './embedUrl';
import htmlVideo from './htmlVideo';

export {
  embedUrl,
  htmlVideo,
};
