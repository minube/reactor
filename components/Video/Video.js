import {
  bool, func, number, string,
} from 'prop-types';
import React, { Component } from 'react';
import { Platform, View } from 'react-native';

import Activity from '../Activity';
import { embedUrl, htmlVideo } from './modules';
import styles from './Video.style';

const isWeb = Platform.OS === 'web';
let WebView;
if (!isWeb) WebView = require('react-native').WebView; // eslint-disable-line

class Video extends Component {
  state = {
    ready: false,
  }

  shouldComponentUpdate({ autoPlay, source }, { ready }) {
    const { el = {}, props, state } = this;

    if (el.play && el.pause) el[autoPlay ? 'play' : 'pause']();
    return (source !== props.source || ready !== state.ready);
  }

  _onLoad = () => {
    const { props: { onLoad } } = this;

    this.setState({ ready: true });
    onLoad();
  }

  render() {
    const {
      _onLoad,
      props: {
        autoPlay, controls, height, onLoad, preload, source, width, ...inherit
      },
      state: { ready },
    } = this;
    const embed = embedUrl(source);
    const dimensions = {
      height, width, maxHeight: height, maxWidth: width,
    };
    const pointerEvents = !controls ? 'none' : undefined;

    return (
      <View
        pointerEvents={!controls && autoPlay ? 'none' : undefined}
        style={[styles.container, dimensions, !ready && styles.loading, inherit.style]}
      >
        { isWeb && !ready && <Activity size="large" style={styles.activity} /> }

        { isWeb && !embed
          && (
          <video
            {...inherit}
            ref={(el) => { this.el = el; }}
            autoPlay={autoPlay}
            controls={controls ? 'true' : undefined}
            onLoadedData={_onLoad}
            pointerEvents={pointerEvents}
            preload={preload ? 'auto' : 'none'}
            style={{ objectFit: 'cover', width, height }}
          >
            <source src={source} />
            <track kind="captions" />
          </video>
          ) }

        { isWeb && embed
          && (
          <iframe
            allowFullScreen
            key={embed}
            frameBorder={0}
            onLoad={_onLoad}
            pointerEvents={pointerEvents}
            width="100%"
            height="100%"
            src={`${embed}&autoplay=${autoPlay ? 1 : 0}`}
            title={embed}
          />
          ) }

        { !isWeb
          && (
          <WebView
            allowsInlineMediaPlayback
            mediaPlaybackRequiresUserAction={false}
            scalesPageToFit={false}
            scrollEnabled={false}
            onLoadStart={onLoad}
            source={{ html: htmlVideo(this.props) }}
            style={[styles.webView, dimensions]}
          />
          ) }
      </View>
    );
  }
}

Video.propTypes = {
  autoPlay: bool,
  controls: bool,
  height: number,
  onLoad: func,
  preload: bool,
  source: string.isRequired,
  width: number,
};

Video.defaultProps = {
  autoPlay: false,
  controls: false,
  height: 180,
  onLoad() {},
  preload: false,
  width: 320,
};

export default Video;
