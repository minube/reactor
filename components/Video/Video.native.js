import {
  bool, func, number, string,
} from 'prop-types';
import React from 'react';
import { View, WebView } from 'react-native';

import { htmlVideo } from './modules';
import styles from './Video.style';

const Video = ({
  onLoad, height, width, ...inherit
}) => {
  const dimensions = {
    height, width, maxHeight: height, maxWidth: width,
  };

  return (
    <View style={[styles.container, dimensions, inherit.style]}>
      <WebView
        allowsInlineMediaPlayback
        mediaPlaybackRequiresUserAction={false}
        scalesPageToFit={false}
        scrollEnabled={false}
        onLoadStart={onLoad}
        source={{ html: htmlVideo({ height, width, ...inherit }) }}
        style={[styles.webView, dimensions]}
      />
    </View>
  );
};

Video.propTypes = {
  autoPlay: bool,
  controls: bool,
  height: number,
  onLoad: func,
  preload: bool,
  source: string.isRequired,
  width: number,
};

Video.defaultProps = {
  autoPlay: false,
  controls: false,
  height: 180,
  onLoad() {},
  preload: false,
  width: 320,
};

export default Video;
