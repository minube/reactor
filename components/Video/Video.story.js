import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean, number, object } from '@storybook/addon-knobs/react';

import LayoutView from '../Layout';
import Video from './Video';

const VIDEO_PLACEHOLDER = 'https://coverr.co/s3/mp4/Cloud_Surf.mp4';
const VIDEO_STYLE = {
  borderRadius: 24,
};

storiesOf('✅ Video', module)
  .addWithJSX('default', () => (
    <Video />
  ))
  .addWithJSX('source', () => (
    <Video source={VIDEO_PLACEHOLDER} />
  ))
  .addWithJSX('youtube source', () => (
    <Video source="https://www.youtube.com/watch?v=cx4MxQcD8Fk" />
  ))
  .addWithJSX('vimeo source', () => (
    <Video source="https://vimeo.com/225434434" />
  ))
  .addWithJSX('autoPlay', () => (
    <Video autoPlay source={VIDEO_PLACEHOLDER} />
  ))
  .addWithJSX('controls', () => (
    <Video controls source={VIDEO_PLACEHOLDER} />
  ))
  .addWithJSX('dimensions', () => (
    <Video height={128} source={VIDEO_PLACEHOLDER} width={256} />
  ))
  .addWithJSX('⚡ onLoad (only web)', () => (
    <Video onLoad={action('Video.onLoad()')} source={VIDEO_PLACEHOLDER} />
  ))
  .addWithJSX('preload', () => (
    <Video preload source={VIDEO_PLACEHOLDER} />
  ))
  .addWithJSX('style', () => (
    <Video source={VIDEO_PLACEHOLDER} style={VIDEO_STYLE} />
  ))
  .addWithJSX('🏀 Playground', () => (
    <LayoutView>
      <Video
        autoplay={boolean('autoplay', true)}
        controls={boolean('controls', false)}
        height={number('height', 226)}
        preload={boolean('preload', false)}
        source={text('source', VIDEO_PLACEHOLDER)}
        style={object('style', VIDEO_STYLE, 'video-style')}
        width={number('width', 428)}
      />
    </LayoutView>
  ));
