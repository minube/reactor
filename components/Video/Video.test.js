import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Video from './Video';

const VIDEO_PLACEHOLDER = 'http://download.blender.org/peach/bigbuckbunny_movies/BigBuckBunny_320x180.mp4';


describe('<Video>', () => {
  it('renders', () => {
    const tree = renderer.create(<Video />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {source}', () => {
    const tree = renderer.create(<Video source={VIDEO_PLACEHOLDER} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {source} (youtube)', () => {
    const tree = renderer.create(<Video source="https://www.youtube.com/watch?v=cx4MxQcD8Fk" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {source} (vimeo)', () => {
    const tree = renderer.create(<Video source="https://vimeo.com/225434434" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {autoplay}', () => {
    const tree = renderer.create(<Video source={VIDEO_PLACEHOLDER} autoplay />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onLoad}', () => {
    // @TODO: Test <Video /> {onLoad}
  });

  it('when {style}', () => {
    const tree = renderer.create(<Video style={{ width: 640, height: 400,}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Video />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

