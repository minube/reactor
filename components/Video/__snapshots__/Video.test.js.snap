// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video> renders 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "height": 180,
        "maxHeight": 180,
        "maxWidth": 320,
        "width": 320,
      },
      Object {
        "backgroundColor": "#f4f4f4",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={true}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "html": "<video
         
      height=\\"100%\\"
      playsinline
      preload=\\"true\\"
      src=\\"undefined\\"
      style=\\"object-fit: cover; background: #CDD2D4; position: absolute; left: 0; top: 0; margin: 0; padding: 0; overflow: hidden;\\"
      webkit-playsinline
      width=\\"100%\\"
    />",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "backgroundColor": "#CDD2D4",
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "height": 180,
              "maxHeight": 180,
              "maxWidth": 320,
              "width": 320,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;

exports[`<Video> when {autoplay} 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "height": 180,
        "maxHeight": 180,
        "maxWidth": 320,
        "width": 320,
      },
      Object {
        "backgroundColor": "#f4f4f4",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={true}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "html": "<video
         
      height=\\"100%\\"
      playsinline
      preload=\\"true\\"
      src=\\"https://storage.googleapis.com/coverr-main/mp4/Surfers-Paradise.mp4\\"
      style=\\"object-fit: cover; background: #CDD2D4; position: absolute; left: 0; top: 0; margin: 0; padding: 0; overflow: hidden;\\"
      webkit-playsinline
      width=\\"100%\\"
    />",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "backgroundColor": "#CDD2D4",
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "height": 180,
              "maxHeight": 180,
              "maxWidth": 320,
              "width": 320,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;

exports[`<Video> when {controls} 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "height": 180,
        "maxHeight": 180,
        "maxWidth": 320,
        "width": 320,
      },
      Object {
        "backgroundColor": "#f4f4f4",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={true}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "html": "<video
       controls  
      height=\\"100%\\"
      playsinline
      preload=\\"true\\"
      src=\\"https://storage.googleapis.com/coverr-main/mp4/Surfers-Paradise.mp4\\"
      style=\\"object-fit: cover; background: #CDD2D4; position: absolute; left: 0; top: 0; margin: 0; padding: 0; overflow: hidden;\\"
      webkit-playsinline
      width=\\"100%\\"
    />",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "backgroundColor": "#CDD2D4",
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "height": 180,
              "maxHeight": 180,
              "maxWidth": 320,
              "width": 320,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;

exports[`<Video> when {dimensions} 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "height": 128,
        "maxHeight": 128,
        "maxWidth": 256,
        "width": 256,
      },
      Object {
        "backgroundColor": "#f4f4f4",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={true}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "html": "<video
         
      height=\\"100%\\"
      playsinline
      preload=\\"true\\"
      src=\\"https://storage.googleapis.com/coverr-main/mp4/Surfers-Paradise.mp4\\"
      style=\\"object-fit: cover; background: #CDD2D4; position: absolute; left: 0; top: 0; margin: 0; padding: 0; overflow: hidden;\\"
      webkit-playsinline
      width=\\"100%\\"
    />",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "backgroundColor": "#CDD2D4",
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "height": 128,
              "maxHeight": 128,
              "maxWidth": 256,
              "width": 256,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;

exports[`<Video> when {preload} 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "height": 180,
        "maxHeight": 180,
        "maxWidth": 320,
        "width": 320,
      },
      Object {
        "backgroundColor": "#f4f4f4",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={true}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "html": "<video
         
      height=\\"100%\\"
      playsinline
      preload=\\"true\\"
      src=\\"https://storage.googleapis.com/coverr-main/mp4/Surfers-Paradise.mp4\\"
      style=\\"object-fit: cover; background: #CDD2D4; position: absolute; left: 0; top: 0; margin: 0; padding: 0; overflow: hidden;\\"
      webkit-playsinline
      width=\\"100%\\"
    />",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "backgroundColor": "#CDD2D4",
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "height": 180,
              "maxHeight": 180,
              "maxWidth": 320,
              "width": 320,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;

exports[`<Video> when {source} (vimeo) 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "height": 180,
        "maxHeight": 180,
        "maxWidth": 320,
        "width": 320,
      },
      Object {
        "backgroundColor": "#f4f4f4",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={true}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "html": "<iframe
      allowtransparency=\\"true\\"
      frameBorder=\\"0\\"
      src=\\"https://player.vimeo.com/video/225434434?title=0&byline=0&portrait=0&autoplay=0\\"
      height=\\"180\\"
      style=\\"background: transparent; position: absolute; left: 0; top: 0; margin: 0; padding: 0; overflow: hidden;\\"
      title=\\"https://player.vimeo.com/video/225434434?title=0&byline=0&portrait=0\\"
      width=\\"320\\"
    />",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "backgroundColor": "#CDD2D4",
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "height": 180,
              "maxHeight": 180,
              "maxWidth": 320,
              "width": 320,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;

exports[`<Video> when {source} (youtube) 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "height": 180,
        "maxHeight": 180,
        "maxWidth": 320,
        "width": 320,
      },
      Object {
        "backgroundColor": "#f4f4f4",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={true}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "html": "<iframe
      allowtransparency=\\"true\\"
      frameBorder=\\"0\\"
      src=\\"https://www.youtube.com/embed/cx4MxQcD8Fk?title=0&autoplay=0\\"
      height=\\"180\\"
      style=\\"background: transparent; position: absolute; left: 0; top: 0; margin: 0; padding: 0; overflow: hidden;\\"
      title=\\"https://www.youtube.com/embed/cx4MxQcD8Fk?title=0\\"
      width=\\"320\\"
    />",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "backgroundColor": "#CDD2D4",
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "height": 180,
              "maxHeight": 180,
              "maxWidth": 320,
              "width": 320,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;

exports[`<Video> when {source} 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "height": 180,
        "maxHeight": 180,
        "maxWidth": 320,
        "width": 320,
      },
      Object {
        "backgroundColor": "#f4f4f4",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={true}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "html": "<video
         
      height=\\"100%\\"
      playsinline
      preload=\\"true\\"
      src=\\"https://storage.googleapis.com/coverr-main/mp4/Surfers-Paradise.mp4\\"
      style=\\"object-fit: cover; background: #CDD2D4; position: absolute; left: 0; top: 0; margin: 0; padding: 0; overflow: hidden;\\"
      webkit-playsinline
      width=\\"100%\\"
    />",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "backgroundColor": "#CDD2D4",
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "height": 180,
              "maxHeight": 180,
              "maxWidth": 320,
              "width": 320,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;

exports[`<Video> when {style} 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "justifyContent": "center",
        "overflow": "hidden",
      },
      Object {
        "height": 180,
        "maxHeight": 180,
        "maxWidth": 320,
        "width": 320,
      },
      Object {
        "backgroundColor": "#f4f4f4",
      },
      Object {
        "height": 400,
        "width": 640,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      allowsInlineMediaPlayback={true}
      automaticallyAdjustContentInsets={undefined}
      bounces={undefined}
      contentInset={undefined}
      dataDetectorTypes={undefined}
      decelerationRate={undefined}
      injectedJavaScript={undefined}
      mediaPlaybackRequiresUserAction={false}
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={undefined}
      scalesPageToFit={false}
      scrollEnabled={false}
      source={
        Object {
          "html": "<video
         
      height=\\"100%\\"
      playsinline
      preload=\\"true\\"
      src=\\"undefined\\"
      style=\\"object-fit: cover; background: #CDD2D4; position: absolute; left: 0; top: 0; margin: 0; padding: 0; overflow: hidden;\\"
      webkit-playsinline
      width=\\"100%\\"
    />",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Array [
            Object {
              "backgroundColor": "#CDD2D4",
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "height": 180,
              "maxHeight": 180,
              "maxWidth": 320,
              "width": 320,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;
